dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -341,6 +354,41 @@ object DistriOptimizer {
     }
   }
 
+  private def saveSummary[T: ClassTag](
+        trainSummary: Option[TrainSummary],
+        models: RDD[Cache[T]],
+        driverState: Table,
+        parameters: AllReduceParameter[T])(implicit ev: TensorNumeric[T]): Unit = {
+    val currentIteration = driverState[Int]("neval") - 1
+    if (trainSummary.isDefined && !trainSummary.get.getTrigger().isEmpty) {
+      val trigger = trainSummary.get.getTrigger()
+      val parametersTrigger = trigger.getOrElse[Trigger]("parameters", null)
+      if (null != parametersTrigger && parametersTrigger(driverState)) {
+        val model = getModelWithGradient(models, parameters)
+        val parametersTable = model.getParametersTable()
+        // Parallelize to create Histogram.
+        Engine.default.invoke(
+          parametersTable.keySet.toSeq.map(moduleName => () => {
+            val paramTable = parametersTable[Table](moduleName)
+            paramTable.keySet.foreach { paramName =>
+              trainSummary.get.addHistogram(
+                s"$moduleName/$paramName", paramTable[Tensor[T]](paramName), currentIteration)}
+          }))
+      }
+      val scalarTrigger = trigger.filter(!_._1.equals("parameters"))
```

Comment:
scalarTriggers?

!_._1.equals("parameters") is hard to understand, provide a method with meaningful name?